test/nlp/test_agtextpredictor.py

```diff
@@ -0,0 +1,139 @@
+            "The tech-laced Nasdaq Composite .IXIC rallied 30.46 points , or 2.04 percent , to 1,520.15 .",
+            "The DVD CCA appealed that decision to the U.S. Supreme Court .",
+            "The Nasdaq had a weekly gain of 17.27 , or 1.2 percent , closing at 1,520.15 on Friday .",
+            "The DVD-CCA then appealed to the state Supreme Court .",  
+            "Yucaipa owned Dominick 's before selling the chain to Safeway in 1998 for $ 2.5 billion .",
+            "They had published an advertisement on the Internet on June 10 , offering the cargo for sale , he added .",
+        ],
+        "numerical1": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
+        "categorical1": ["a", "b", "a", "a", "a", "b", "a", "a", "a", "b", "a", "a"],
+        "label": [1, 0, 2, 0, 1, 2, 0, 1, 1, 2, 0, 1],
+    }
+    train_dataset = pd.DataFrame(train_data)
+
+    test_data = {
+            "sentence1": [
+                "That compared with $ 35.18 million , or 24 cents per share , in the year-ago period .",
+                "Shares of Genentech , a much larger company with several products on the market , rose more than 2 percent .",
+                "Legislation making it harder for consumers to erase their debts in bankruptcy court won overwhelming House approval in March .",
+                "The Nasdaq composite index increased 10.73 , or 0.7 percent , to 1,514.77 .",
+            ],
+            "sentence2": [
+                "Earnings were affected by a non-recurring $ 8 million tax benefit in the year-ago period .",
+                "Shares of Xoma fell 16 percent in early trade , while shares of Genentech , a much larger company with several products on the market , were up 2 percent .",
+                "Legislation making it harder for consumers to erase their debts in bankruptcy court won speedy , House approval in March and was endorsed by the White House .",
+                "The Nasdaq Composite index , full of technology stocks , was lately up around 18 points .",
+            ],
+            "numerical1": [3, 4, 5, 6],
+            "categorical1": ["b", "a", "a", "b"],
+            "label": [0, 1, 1, 2],
+    }
+    test_dataset = pd.DataFrame(test_data)
+
+    # FORCE THE SAME TRAIN-VALID SPLIT IN & OUT THE PREDICTOR
+    holdout_frac = default_holdout_frac(len(train_dataset), False)
+
+    _, valid_dataset = train_test_split(train_dataset,
+                                    test_size=holdout_frac,
+                                    random_state=np.random.RandomState(seed))
+    
+    feature_columns = ["sentence1", "sentence2", "numerical1", "categorical1"]
+
+    automl = AutoML()
+    automl_settings = {
+        "gpu_per_trial": 0,
+        "max_iter": 2,
+        "time_budget": 50,
+        "task": "mm_multi",
+        "metric": "accuracy",
+    }
+
+    automl_settings["custom_fix_args"] = {
```

Comment:
Please define a class to flaml/nlp/utils.py:class AGArgs to address the default value of AutoGluon, that the user does not need to see. Move any variable here to AGArgs so you don't need to set the value here.

Reply:
Will do. Do I need any function like `_init_hpo_args()` in the TransformersEstimator then?